### PR TITLE
Live location sharing - basic maximised beacon map

### PR DESCRIPTION
**Pull request by kerryarchibald**
With: https://github.com/matrix-org/matrix-js-sdk/pull/2296

Includes: #8316

Adds a basic maximised view for live beacons.
Just centers on first beacon in list for now. Zoom/centering logic coming in new PR

https://user-images.githubusercontent.com/3055605/163200636-0ec327ae-ad31-4946-a616-5a1e029337c8.mov

Screenshot 2022-04-13 at 16 12 35

---
Here's what your changelog entry will look like:

## ✨ Features
 * Live location sharing - basic maximised beacon map ([\#8310](https://github.com/matrix-org/matrix-react-sdk/pull/8310)). Contributed by @kerryarchibald.

Preview: https://pr8310--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/beacon/BeaconViewDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        >
+            <MatrixClientContext.Provider value={matrixClient}>
+                <Map
+                    id='mx_BeaconViewDialog'
+                    centerGeoUri={mapCenterUri}
+                    interactive
+                    className="mx_BeaconViewDialog_map"
+                >
+                    {
+                        ({ map }: { map: maplibregl.Map}) =>
```
Comment (turt2live) [inline src/components/views/beacon/BeaconViewDialog.tsx:70]: I may be missing something very obvious, but this looks like a function which isn't called?

File: test/components/views/beacon/BeaconMarker-test.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    const roomId = '!room:server';
+    const aliceId = '@alice:server';
+
+    const aliceMember = new RoomMember(roomId, aliceId);
+
+    const mockMap = new maplibregl.Map();
+
+    const mockClient = getMockClientWithEventEmitter({
+        getClientWellKnown: jest.fn().mockReturnValue({
+            "m.tile_server": { map_style_url: 'maps.com' },
```
Comment (turt2live) [inline test/components/views/beacon/BeaconMarker-test.tsx:46]: I've mentioned this elsewhere, and apologies for only noticing it now, but `m.tile_server` is an illegal value in the spec so this needs to be renamed. A different PR can take care of that.

**Review by turt2live — APPROVED** [review]
otherwise lgtm

File: test/components/views/beacon/BeaconMarker-test.tsx (hunk 2 above)
Comment (kerryarchibald) [reply test/components/views/beacon/BeaconMarker-test.tsx:46]: oh, good catch! opened a PR to rename https://github.com/matrix-org/matrix-react-sdk/pull/8316

File: src/components/views/beacon/BeaconViewDialog.tsx (hunk 1 above)
Comment (kerryarchibald) [reply src/components/views/beacon/BeaconViewDialog.tsx:70]: this is a render prop / 'function as a child', it's called by `Map` with the map instance once it is created.

**Pull request merged**